packages/polling-controller/src/PollingController.test.ts

```diff
@@ -0,0 +1,214 @@
+import { ControllerMessenger } from '@metamask/base-controller';
+
+import type { PollingCompleteType } from './PollingController';
+import PollingController from './PollingController';
+
+describe('PollingController', () => {
+  let executePollMock: PollingController<any, any, any>['executePoll'];
+
+  beforeEach(() => {
+    executePollMock = jest.fn().mockImplementation(async () => {
```

Comment:
Nit: Typically we recommend using setup functions over beforeEach where possible. e.g. in this case, we could define a `createExecutePollMock` function and use that in each test directly instead. Generally setup functions make for easier-to-read tests and avoids maintenance problems later if we add tests that this mock isn't suitable for.

More on that here: https://github.com/MetaMask/contributor-docs/blob/main/docs/unit-testing.md#avoid-the-use-of-beforeeach